### PR TITLE
refactor(helm): add helper function to add quotes to annotation_match

### PR DESCRIPTION
**Pull request by kkujawa-sumo**
quote function is not use as it automatically adds `\` before special characters

Example tested configuration:

values.yaml
```yaml
fluentd:
  logs:
    containers:
      perContainerAnnotationsEnabled: true
      perContainerAnnotationPrefixes:
        - "sumologic.com/"
        - "tailing-sidecar.sumologic.com/"
        
  metadata:
   annotation_match:
    - 'sumologic\.com.*'
    - 'tailing-sidecar.sumologic\.com.*'
    - "test-sumologic\\.com.*"
    - "test-tailing-sidecar.sumologic\\.com.*"
```

pod1.yaml
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: pod1
  namespace: sumologic
  annotations:
    sumologic.com/count1.sourceCategory: test-source-category51
    sumologic.com/count2.sourceCategory: test-source-category61
  labels:
    app: pod-with-annotations
spec:
  containers:
  - name: count1
    image: busybox
    args:
    - /bin/sh
    - -c
    - >
      i=0;
      while true;
      do
        echo "example0: $i $(date)"
        echo "example1: $i $(date)"
        echo "example2: $i $(date)"
        i=$((i+1));
        sleep 5;
      done
    volumeMounts:
    - name: varlog
      mountPath: /var/log
    - name: varlogconfig
      mountPath: /varconfig/log
  - name: count2
    image: busybox
    args:
    - /bin/sh
    - -c
    - >
      i=0;
      while true;
      do
        echo "example0: $i $(date)"
        echo "example1: $i $(date)"
        echo "example2: $i $(date)"
        i=$((i+1));
        sleep 5;
      done
    volumeMounts:
    - name: varlog
      mountPath: /var/log
    - name: varlogconfig
      mountPath: /varconfig/log
  volumes:
  - name: varlog
    emptyDir: {}
  - name: varlogconfig
    emptyDir: {}
```

pod2.yaml
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: pod2
  namespace: sumologic
  annotations:
    tailing-sidecar.sumologic.com/count1.sourceCategory: test-source-category31
    tailing-sidecar.sumologic.com/count2.sourceCategory: test-source-category41
  labels:
    app: pod-with-annotations
spec:
  containers:
  - name: count1
    image: busybox
    args:
    - /bin/sh
    - -c
    - >
      i=0;
      while true;
      do
        echo "example0: $i $(date)"
        echo "example1: $i $(date)"
        echo "example2: $i $(date)"
        i=$((i+1));
        sleep 5;
      done
    volumeMounts:
    - name: varlog
      mountPath: /var/log
    - name: varlogconfig
      mountPath: /varconfig/log
  - name: count2
    image: busybox
    args:
    - /bin/sh
    - -c
    - >
      i=0;
      while true;
      do
        echo "example0: $i $(date)"
        echo "example1: $i $(date)"
        echo "example2: $i $(date)"
        i=$((i+1));
        sleep 5;
      done
    volumeMounts:
    - name: varlog
      mountPath: /var/log
    - name: varlogconfig
      mountPath: /varconfig/log
  volumes:
  - name: varlog
    emptyDir: {}
  - name: varlogconfig
    emptyDir: {}
```

fluentd configuration in helm templates (`helm --dry-run`)
```
  logs.kubernetes.metadata.filter.conf: |-
    annotation_match '["sumologic\.com.*","tailing-sidecar.sumologic\.com.*","test-sumologic\.com.*","test-tailing-sidecar.sumologic\.com.*"]'
    de_dot false
    watch "true"
    ca_file ""
    verify_ssl "true"
    client_cert ""
    client_key ""
    bearer_token_file ""
    cache_size "10000"
    cache_ttl "7200"
    tag_to_kubernetes_name_regexp '.+?\.containers\.(?<pod_name>[^_]+)_(?<namespace>[^_]+)_(?<container_name>.+)-(?<docker_id>[a-z0-9]{64})\.log$'
```

fluentd configuration captured by fluentd pod (`kubectl logs collection-sumologic-fluentd-logs-2`)

```
    <filter containers.**>
      @type kubernetes_metadata
      @log_level "error"
      annotation_match ["sumologic.com.*","tailing-sidecar.sumologic.com.*","test-sumologic.com.*","test-tailing-sidecar.sumologic.com.*"]
      de_dot false
      watch true
      ca_file ""
      verify_ssl true
      client_cert ""
      client_key ""
      bearer_token_file ""
      cache_size 10000
      cache_ttl 7200
      tag_to_kubernetes_name_regexp ".+?\\.containers\\.(?<pod_name>[^_]+)_(?<namespace>[^_]+)_(?<container_name>.+)-(?<docker_id>[a-z0-9]{64})\\.log$"
    </filter>
```

### CODE REVIEW

File: deploy/helm/sumologic/templates/_helpers.tpl [hunk 1]
```diff
@@ -982,3 +982,14 @@ Example:
 {{- define "kubernetes.minor" -}}
 {{- print (regexFind "^\\d+" .Capabilities.KubeVersion.Minor) -}}
 {{- end -}}
+
+{{- define "fluentd.metadata.annotations_match.quotes" -}}
+{{- $matches_with_quotes := list -}}
+{{- range $match := .Values.fluentd.metadata.annotation_match  }}
+{{- $match_with_quotes := printf "\"%s\"" $match }}
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/templates/_helpers.tpl:989]: please use `quote` function as someone potentially can put quotes in variable

Comment (sumo-drosiek) [reply deploy/helm/sumologic/templates/_helpers.tpl:989]: https://github.com/Masterminds/sprig/blob/master/docs/strings.md#quote-and-squote

Comment (kkujawa-sumo) [reply deploy/helm/sumologic/templates/_helpers.tpl:989]: I've tested with `quote` and I noticed that it adds `\` before special characters. I used this in following way:
```
{{- $match_with_quotes := $match  | quote }}
```
and then in help templates there was following form:
```
annotation_match ["sumologic\\.com.*","tailing-sidecar.sumologic\\.com.*","test-sumologic\\.com.*","test-tailing-sidecar.sumologic\\.com.*"]
```
and in fluend logs I saw that Pod had following configuration
```
 <filter containers.**>
      @type kubernetes_metadata
      @log_level "error"
      annotation_match ["sumologic\\.com.*","tailing-sidecar.sumologic\\.com.*","test-sumologic\\.com.*","test-tailing-sidecar.sumologic\\.com.*"]
      de_dot false
      watch true
      ca_file ""
      verify_ssl true
      client_cert ""
      client_key ""
      bearer_token_file ""
      cache_size 10000
      cache_ttl 7200
      tag_to_kubernetes_name_regexp ".+?\\.containers\\.(?<pod_name>[^_]+)_(?<namespace>[^_]+)_(?<container_name>.+)-(?<docker_id>[a-z0-9]{64})\\.log$"
    </filter>
```

**Review by sumo-drosiek — APPROVED** [review]
Thanks!

**Pull request merged**